.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,102 @@
+name: build
+on:
+  push:
+    branches:
+      - master
+jobs:
+  benchmark:
+    name: Benchmark ${{ matrix.project }} | PARALLEL= ${{ matrix.parallel }}
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    strategy:
+      matrix:
+        project: [java_simple, java_maven, java_gradle]
+        parallel: ['parallel', 'serial', 'multithreaded']
+
+    env:
+      PARALLEL: ${{ matrix.parallel }}
+      BENCHMARK_OUT_DIR: "../out"
+
+    steps:
+      - name: Set up Go 1.13
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.13
+        id: go
+
+      - name: Setup java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: gauge
+
+      - name: Install Gauge
+        run: |
+          cd gauge
+          go run build/make.go && go run build/make.go --install --prefix=/tmp/gauge --verbose
+          echo "::add-path::/tmp/gauge/bin"
+
+      - name: Install gauge-java
+        run: gauge install java
+
+      - name: Prep benchmark
+        uses: actions/checkout@v2
+        with:
+          repository: getgauge/gauge-benchmarks
+          path: gauge-benchmarks
+          ref: master
+
+      - name: Run benchmark ${{matrix.project}} | ${{matrix.parallel}}
+        env:
+          PARALLEL: ${{ matrix.parallel != 'serial' }}
+          enable_multithreading: ${{ matrix.parallel == 'multithreaded' }}
+        run: |
+          mkdir -p gauge-benchmarks/out
+          cd gauge-benchmarks/${{matrix.project}}
+          ./run.sh
+
+      - name: Upload benchmark
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{matrix.project}}_${{matrix.parallel}}
+          path: gauge-benchmarks/out
+
+  publish-benchmark:
+    name: Publish Benchmark
+    runs-on: ubuntu-latest
+    needs: [ benchmark ]
+    if: github.event_name == 'pull_request'
+
+    env:
+      FOO: ${{ secrets.FOO }}
+    steps:
+      - name: Test Org Secret
+        run: |
+          echo "FOO: $FOO"
+
```

Comment:
Is this needed, or just part of debugging?

Reply:
this was part of debugging. fixed.